### PR TITLE
chore: deploy eth gateway via fluence cli

### PR DESCRIPTION
**Pull request by folex**

### CODE REVIEW

**Review by boneyard93501 — COMMENTED** [review]
fluence deploy of ipfs_package throws an error:

```bash
<...>
? Do you want to deploy all of these services? Yes
Going to upload module: ipfs_adapter
Going to upload module: ipfs_cli
Module 'ipfs_adapter' was uploaded
Module 'ipfs_cli' was uploaded
Now time to make the blueprint...
[marine service "avm"]: call failed with an error `Local service error, ret_code is 1, error message is '"Error: engine error: Wasmer error: link error: Import not found, namespace: host, name: ipfs\nEngine(MarineError(EngineError(WasmerError(\"link error: Import not found, namespace: host, name: ipfs\"))))"'`, peerId `12D3KooWKnEqMfYo9zvfHmqTLpLdiHXPe4SVqUWcWHDJdFGrSmcA`[marine service "avm"]: xor caught an error: Local service error, ret_code is 1, error message is '"Error: engine error: Wasmer error: link error: Import not found, namespace: host, name: ipfs\nEngine(MarineError(EngineError(WasmerError(\"link error: Import not found, namespace: host, name: ipfs\"))))"'
<...>
Deploying:
  service: ipfs_package
  deployId: default
  on: 12D3KooWKnEqMfYo9zvfHmqTLpLdiHXPe4SVqUWcWHDJdFGrSmcA... failed
 ›   Warning: Wasn't able to deploy service
 ›   Error: process exited with code 1
 ›   [marine service "avm"]: call failed with an error `Local service error, ret_code is 1, error message
 ›   is '"Error: engine error: Wasmer error: link error: Import not found, namespace: host, name:
 ›   ipfs\nEngine(MarineError(EngineError(WasmerError(\"link error: Import not found, namespace: host,
 ›   name: ipfs\"))))"'`, peerId `12D3KooWKnEqMfYo9zvfHmqTLpLdiHXPe4SVqUWcWHDJdFGrSmcA`[marine service
 ›   "avm"]: xor caught an error: Local service error, ret_code is 1, error message is '"Error: engine
 ›   error: Wasmer error: link error: Import not found, namespace: host, name:
 ›   ipfs\nEngine(MarineError(EngineError(WasmerError(\"link error: Import not found, namespace: host,
 ›   name: ipfs\"))))"'[marine service "avm"]: call failed with an error `Local service error, ret_code is
 ›   1, error message is '"Error: engine error: Wasmer error: link error: Import not found, namespace:
 ›   host, name: ipfs\nEngine(MarineError(EngineError(WasmerError(\"link error: Import not found,
 ›   namespace: host, name: ipfs\"))))"'`, peerId
 ›   `12D3KooWKnEqMfYo9zvfHmqTLpLdiHXPe4SVqUWcWHDJdFGrSmcA`[marine service "avm"]: xor caught an error:
 ›   Local service error, ret_code is 1, error message is '"Error: engine error: Wasmer error: link error:
 ›   Import not found, namespace: host, name: ipfs\nEngine(MarineError(EngineError(WasmerError(\"link
 ›   error: Import not found, namespace: host, name: ipfs\"))))"'[marine service "avm"]:   waiting for an
 ›   argument with name 'service_id'[marine service "avm"]:   waiting for an argument with name
 ›   'service_id'[marine service "avm"]:   waiting for an argument with name 'service_id'Local service
 ›   error, ret_code is 1, error message is '"Error: engine error: Wasmer error: link error: Import not
 ›   found, namespace: host, name: ipfs\nEngine(MarineError(EngineError(WasmerError(\"link error: Import
 ›   not found, namespace: host, name: ipfs\"))))"'
 ›
 ›   Try 'aqua --help' for usage instructions
 ›   Debug info:
 ›   /Users/bebo/.fluence/npm/@fluencelabs/aqua/0.9.1-374/node_modules/.bin/aqua remote deploy_service
 ›   --config-path /Users/bebo/localdev/raw-examples/aqua-examples/decentralized-blockchain-gateway/.fluenc
 ›   e/tmp/deploy.json --service default --addr
 ›   /dns4/kras-07.fluence.dev/tcp/19001/wss/p2p/12D3KooWEFFCZnar1cUJQ3rMWjvPQg6yMV2aXWs2DkJNSRbduBWn --sk
 ›   <SECRET_KEY_IS_HIDDEN>
 ›
Going to upload module: curl_adapter
Going to upload module: multi_provide
<...>
```

File: aqua-examples/decentralized-blockchain-gateway/scripts/build.sh [hunk 1]
```diff
@@ -8,33 +8,31 @@ rm -f artifacts/*.wasm
...
 cp target/wasm32-wasi/release/ipfs_cli.wasm ../../artifacts/
 
 cd ../utilities
-cargo update --aggressive
 marine build --release
 cp target/wasm32-wasi/release/utilities.wasm ../../artifacts/
 
+cd ../eth-rpc
+marine build --release
+cp target/wasm32-wasi/release/eth-rpc.wasm ../../artifacts/
```
Comment (boneyard93501) [inline aqua-examples/decentralized-blockchain-gateway/scripts/build.sh:36]: should be  .../eth_rpc.wasm

File: aqua-examples/decentralized-blockchain-gateway/fluence.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# Defines Fluence Project, most importantly - what exactly you want to deploy and how. You can use `fluence init` command to generate a template for new Fluence project
+
+# Documentation: https://github.com/fluencelabs/fluence-cli/tree/main/docs/configs/fluence.md
+
+version: 2
+aquaInputPath: src/aqua/main.aqua
+dependencies:
+  npm:
+    "@fluencelabs/aqua-lib": 0.6.0
+    "@fluencelabs/spell": 0.0.1
```
[the remark this answers is not in the document]
Comment (folex) [reply aqua-examples/decentralized-blockchain-gateway/fluence.yaml:12]: yeah, it's not yet integrated with spells in any way though 🙈

File: aqua-examples/decentralized-blockchain-gateway/configs/deployment_cfg.json [hunk 3]
```diff
@@ -52,5 +52,22 @@
         "logger_enabled": true
       }
     ]
+  },
+  "eth-rpc": {
+    "modules": [
+      {
+        "name": "curl_adapter",
+        "path": "./artifacts/curl_adapter.wasm",
+        "max_heap_size": "2 MiB",
```
[the remark this answers is not in the document]
Comment (folex) [reply aqua-examples/decentralized-blockchain-gateway/configs/deployment_cfg.json:61]: fluence cli doesn't allow me to omit this :((

File: aqua-examples/decentralized-blockchain-gateway/wasm-modules/multi-provider-query/src/main.rs [hunk 4]
```diff
@@ -1,159 +1,12 @@
...
-        stderr: serde_json::to_string(&response.error).unwrap(),
-    }
-}
+pub fn main() {}
 
 #[marine]
-#[link(wasm_import_module = "curl_adapter")]
-extern "C" {
-    pub fn curl_request(cmd: Vec<String>) -> MountedBinaryResult;
+pub fn greeting(name: String) -> String {
```
[the remark this answers is not in the document]
Comment (folex) [reply aqua-examples/decentralized-blockchain-gateway/wasm-modules/multi-provider-query/src/main.rs:10]: dunno how it got there O_O

File: aqua-examples/decentralized-blockchain-gateway/scripts/build.sh (hunk 1 above)
Comment (folex) [reply aqua-examples/decentralized-blockchain-gateway/scripts/build.sh:36]: `-` vs `_` gives me a headache

**Pull request merged**